dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -329,6 +330,8 @@ public Tool refreshTool(final long toolId, final Long userId, final UserDAO user
             toolTags = getTagsDockerHub(tool);
         } else if (tool.getRegistry().equals(Registry.GITLAB.getDockerPath())) {
             toolTags = getTagsGitLab(tool);
+        } else if (tool.getRegistry().equals((Registry.GITHUB_CONTAINER_REGISTRY.getDockerPath()))) {
```

Comment:
question, not sure if worth a code comment on two issues
1) before this was added, it seems line 336 would go to `getTags` which is implemented by `QuayImageRegistry` and `ManualRegistry`, is that correct?
2) `getTagsGitHubContainerRegistry` can only get image information for tool versions that are already known, but `getTagsDockerHub` and `getTagsGitLab` can create new tags (and versions) based on what their APIs are saying, is that correct?

Reply:
> before this was added, it seems line 336 would go to getTags which is implemented by QuayImageRegistry and ManualRegistry, is that correct?

That's correct. Without this PR, a GHCR tool would go to `ManualRegistry.getTags`.

> getTagsGitHubContainerRegistry can only get image information for tool versions that are already known,

This is correct. If a user adds a tag in the UI, then this tag will be a known tool version and on refresh, the image info for that tool version is retrieved. 

> but getTagsDockerHub and getTagsGitLab can create new tags (and versions) based on what their APIs are saying

This is half correct. These two methods do create new tags based on what their APIs are saying (their APIs show all tags that exist for the image). However, not all of these new tags are actually saved to the database. We only update image info for existing tags on refresh and we do not add new versions for new tags that we may see from the APIs. The actual image info update occurs on line 574 and if you look at that method, it only updates existing tags. 

This was the main factor in my decision to only grab the known tool versions for GHCR tools. I believe it is possible to figure out all the tags for an image using the Docker Registry API, but it's more work and in this case, it seems unnecessary since we don't create new versions for new tags that we see.